File: C++/Data Structure/Stack.cpp
#include<iostream>
using namespace std ;

struct node
{
    int data ;
    struct node* next ;
};

typedef struct node NODE ;
typedef struct node* PNODE ;

class Stack
{
    public:
        PNODE head ;
        int iCountNode ;

        Stack()
        {
            head = NULL ;
            iCountNode = 0 ;
        }

        int CountNode()
        {
            return iCountNode ;
        }

        void Display()
        {
            if(head == NULL)
            {
                cout<<"Stack is empty !!\n";
                return ;
            }
            PNODE temp = NULL ;
            temp = head ;

            while(temp != NULL)
            {
                cout<<"| "<<temp->data<<" |\n";
                temp = temp->next ;
            }
        }

        void Push(int iNo)
        {
            PNODE newn = NULL ;

            newn = new (NODE);

            newn->next = NULL ;
            newn->data = iNo ;

            if(head == NULL)
            {
                head = newn ;
            }
            else
            {
                newn->next = head ;
                head = newn ;
            }
            iCountNode++ ;
        }

        int Pop()
        {
            int no = -1 ;

            PNODE target = NULL ;

            if(head == NULL)
            {
                cout<<"Stack is already empty ! enable to delete element !!\n";
                return -1 ;
            }
            else if(head->next == NULL)
            {
                target = head ;

                no = head->data ;

                delete target ;

                head = NULL ;
            }
            else
            {   
                no = head->data ;

                target = head ;

                head = head->next ;

                delete target ;
            }
            iCountNode-- ;

            return no ;
        }
};

int main()
{
    int iNo = 0 , iOption = 0 , iRetCountNode = 0 , iRetPopelement = 0 ;

    Stack sobj ;

    cout<<"<-- You are using Stack module -->\n";
    cout<<"--------------------------------------------------------\n";
    while(1)
    {
        cout<<"______________________________________________________\n";
        cout<<"1 Display the Stack\n";
        cout<<"2 Count the total elements from the Stack\n";
        cout<<"3 Push the element into the Stack\n";
        // Push and pop is done from the same end.
        cout<<"4 pop the element from thr Stack\n";
        cout<<"5 Exit the Application\n";
        cout<<"______________________________________________________\n";
        cout<<"Enter your option\n";
        cin>>iOption;
        cout<<"______________________________________________________\n";

        if(iOption == 1)
        {
            sobj.Display();
        }
        else if(iOption == 2)
        {
            iRetCountNode = sobj.CountNode();
            cout<<"Total elements in stack are : "<<iRetCountNode<<"\n";
        }
        else if(iOption == 3)
        {
            cout<<"Enter the data : \n";
            cin>>iNo ;

            sobj.Push(iNo);
        }
        else if(iOption == 4)
        {
            iRetPopelement = sobj.Pop();
            cout<<"Poped element is : "<<iRetPopelement<<"\n";
        }
        else if(iOption == 5)
        {
            cout<<"Thankyou for using your module\n";
            break;
        }
        else
        {
            cout<<"!! Invalid Option !!\n";
        } 
    }
}

File: JAVA/Data Structure/Stack.java
import java.util.Scanner;

class Node
{
    int data ;
    Node next ;

    public Node(int iNo)
    {
        data = iNo ;
        next = null ;
    }
}

class Stackk
{
    // characterstics 
    public Node head ;
    public int iCountNode ;

    // constructor
    public Stackk()
    {
        head = null ;
        iCountNode = 0 ;
    }

    // countNode method is used for count the total nodes of stack
    public int countNode()
    {
        if(head == null)
        {
            return 0 ;
        }

        return iCountNode ;
    }

    // displayStack method is used for display the elements which are present inside the Stack
    public void displayStack()
    {
        if(head == null)
        {
            System.out.println("NULL : ! Stack is empty !");
            return ;
        }

        System.out.println("------ Stack (Top to Bottom) ------");
        Node temp = head;

        while(temp != null)
        {
            System.out.println("| " + temp.data + " |");
            temp = temp.next;
        }
        System.out.println("-------- Bottom of Stack ----------");
    }

    // Push method is used for insert the new node inside the Stack
    public void push(int iNo)
    {
        Node newn = new Node(iNo);
        newn.next = head;
        head = newn;

        iCountNode++;

        System.out.println("Data inserted successfully.");
    }

    // pop method is used to Delete the node of Stack
    public int pop()
    {
        int no = -1;

        if (head == null)
        {
            System.out.println("Unable to delete because stack is already empty!!");
            return no;
        }

        no = head.data;
        head = head.next;

        iCountNode--;

        System.out.println("Data deleted successfully.");
        return no;
    }

    // emptystack method is used to delete all nodes and make Stack empty
    public void emptyStack()
    {
        if(head == null)
        {
            System.out.println("Unable to delete because Stack is empty !!");
            return ;
        }

        Node temp = head ;
        Node nextnode = null ;

        while(temp != null)
        {
            nextnode = temp.next ;
            temp.next = null ;

            temp = nextnode ;
        }

        head = null ;
        iCountNode = 0 ;

        System.out.println("All Data deleted successfully, Now Stack is empty");
    }
}

class Stack
{
    public static void main(String arg[])
    {
        Scanner sobj = new Scanner(System.in);

        int iRet = 0 , iOption = 0 , iData = 0 , iDeleteData = 0 ;

        Stackk stack = new Stackk();

        while(true)
        {
            System.out.println("-----------------------------------------------------------------");
            System.out.println("0 : Terminate the Application");
            System.out.println("1 : Display the stack ");
            System.out.println("2 : Count the total nodes of stack");
            System.out.println("3 : push the data (insert the data)");
            System.out.println("4 : pop the data (delete the data)");
            System.out.println("5 : Delete the all data from stack (OR) make stack empty");
            System.out.println("-----------------------------------------------------------------");
            System.out.println("Enter the option : ");
            iOption = sobj.nextInt();
            System.out.println("-----------------------------------------------------------------");

            switch(iOption)
            {
                case 0 :
                    System.out.println("Module is closing");
                    sobj.close();
                    return ;

                case 1 :
                    stack.displayStack();
                    break ;
                
                case 2 :
                    iRet = stack.countNode(); 
                    System.out.println("Total nodes in Stack are : "+iRet);
                    break ;

                case 3 :
                    System.out.println("Enter the data : ");
                    iData = sobj.nextInt();

                    stack.push(iData);
                    break ;

                case 4 :
                    iDeleteData = stack.pop();
                    System.out.println("\n poped element is : "+ iDeleteData + "\n");
                    break ;

                case 5 :
                    stack.emptyStack();
                    break ;

                default :
                    System.out.println("!! Invalid Option !!");
                    break ;
            }     
        }
    }
}